import-export-cli/integration/apim/apim.go

```diff
@@ -2295,3 +2297,178 @@ func getTokenURL(host string, offset int) string {
+
+	policiesURL := instance.publisherRestURL + "/operation-policies/" + policyId
+
+	request := base.CreateGet(policiesURL)
+
+	base.SetDefaultRestAPIHeaders(instance.accessToken, request)
+
+	base.LogRequest("apim.GetAPIPolicy()", request)
+
+	response := base.SendHTTPRequest(request)
+
+	defer response.Body.Close()
+
+	base.ValidateAndLogResponse("apim.GetAPIPolicy()", response, 200)
+
+	json.NewDecoder(response.Body).Decode(&policyResponse)
+
+	return policyResponse
+}
+
+// GetAPIPolicyID : Get API Policy Id using policy name and version from APIM
+func (instance *Client) GetAPIPolicyID(t *testing.T, policyName, policyVersion string) string {
+	var policyListResponse APIPoliciesList
+
+	queryParams := "name:" + policyName + " version:" + policyVersion
+
+	getPoliciesURL := instance.publisherRestURL + "/operation-policies"
+
+	request := base.CreateGet(getPoliciesURL)
+
+	base.SetDefaultRestAPIHeaders(instance.accessToken, request)
+
+	values := url.Values{}
+	values.Add("query", queryParams)
+
+	request.URL.RawQuery = values.Encode()
+
+	base.SetDefaultRestAPIHeaders(instance.accessToken, request)
+
+	base.LogRequest("apim.GetAPIPolicyID()", request)
+
+	response := base.SendHTTPRequest(request)
+
+	defer response.Body.Close()
+
+	base.ValidateAndLogResponse("apim.GetAPIPolicyID()", response, 200)
+
+	err := json.NewDecoder(response.Body).Decode(&policyListResponse)
+	if err != nil {
+		t.Fatal(err)
+	}
+	apiPolicyList := policyListResponse.List
+	return apiPolicyList[0].Id
+}
+
+// AddAPIPolicy : Add new API Policy of different policy types to APIM
+func (instance *Client) AddAPIpolicy(t *testing.T, policySpec []byte, synapseDefFilePath string) interface{} {
+	var apiPolicyResponse interface{}
+
+	apiPolicyURL := instance.publisherRestURL + "/operation-policies"
+
+	body := &bytes.Buffer{}
+	writer := multipart.NewWriter(body)
+
+	synapseDefFile, err := os.Open(synapseDefFilePath)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer synapseDefFile.Close()
+
+	part, err := writer.CreateFormFile("synapsePolicyDefinitionFile", filepath.Base(synapseDefFile.Name()))
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = io.Copy(part, synapseDefFile)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	part, err = writer.CreateFormField("policySpecFile")
+	if err != nil {
+		t.Fatal(err)
+	}
+	part.Write(policySpec)
+
+	err = writer.Close()
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	fmt.Println("Body: ", body)
+
+	request := base.CreatePost(apiPolicyURL, body)
+
+	base.SetDefaultRestAPIHeadersToConsumeFormData(instance.accessToken, request)
+
+	fmt.Println("Request: ", request.Header)
+
+	fmt.Println("Request Content: ", request.Body)
+
+	base.LogRequest("apim.AddAPIPolicy()", request)
+
+	response := base.SendHTTPRequest(request)
+
+	defer response.Body.Close()
+
+	fmt.Println("Response Code: ", response.StatusCode)
+
+	base.ValidateAndLogResponse("apim.AddAPIPolicy()", response, 201)
+
+	json.NewDecoder(response.Body).Decode(&apiPolicyResponse)
+
```

Comment:
We should regsiter a t.Cleanup function when adding artifacts, because when the test run is over, these added artifacts should be deleted from the APIM instance. Refer here [1] for an example.

[1] https://github.com/wso2/product-apim-tooling/blob/7a352c13ca647dddda3bf2004cf9f6d9be95e5ff/import-export-cli/integration/apim/apim.go#L522

Reply:
Fixed by [59b2698](https://github.com/wso2/product-apim-tooling/pull/925/commits/59b2698c4e9aceab8b212ba2afd761955611f7e4)